File: examples/csharp/Tutorial35.cs
/* --------------------------------------------------------------------
 * Tutorial 35
 * 
 * This tutorial shows how to import Excel sheet to DataSet in C#.
 * The data is imported from a specific Excel sheet (For this example 
 * we use the Excel file generated in Tutorial 09).
 * ----------------------------------------------------------------- */

using System;
using System.IO;
using System.Data;
using EasyXLS;

class Tutorial35
{
	
	[STAThread]
	static void Main()
	{
		Console.WriteLine("Tutorial 35\n-----------\n");

        // Create an instance of the class that imports Excel files
		ExcelDocument workbook = new ExcelDocument();

        // Import Excel sheet to DataSet
		Console.WriteLine("Reading file C:\\Samples\\Tutorial09.xlsx.\n");
		DataSet ds = workbook.easy_ReadXLSXSheet_AsDataSet("C:\\Samples\\Tutorial09.xlsx", "First tab");

        // Display imported DataSet values
		DataTable dt = ds.Tables[0];
		for (int row=0; row<dt.Rows.Count; row++)
			for (int column=0; column<dt.Columns.Count; column++)
				Console.WriteLine("At row " + (row + 1) + ", column " + (column + 1) +
					" the value is '" + dt.Rows[row].ItemArray[column] + "'");
 
		Console.Write("\nPress Enter to continue...");
		
		// Dispose memory
		workbook.Dispose();

		Console.ReadLine();
	}
}

File: examples/cpp/Tutorial35.cpp
/* --------------------------------------------------------------------
 * Tutorial 35
 * 
 * This tutorial shows how to import Excel sheet to List in C++.
 * The data is imported from a specific Excel sheet (For this example
 * we use the Excel file generated in Tutorial 09).
 * ----------------------------------------------------------------- */

#include "EasyXLS.h"
#include <conio.h>


int main()
{
	printf("Tutorial 35\n----------\n");

	HRESULT hr;

	// Initialize COM
	hr = CoInitialize(0);

	// Use the SUCCEEDED macro and get a pointer to the interface
	if(SUCCEEDED(hr))
	{
		// Create a pointer to the interface that imports Excel files
		EasyXLS::IExcelDocumentPtr workbook;
		hr = CoCreateInstance(__uuidof(EasyXLS::ExcelDocument),
		NULL,
		CLSCTX_ALL,
		__uuidof(EasyXLS::IExcelDocument),
		(void**) &workbook) ;

		if(SUCCEEDED(hr)){

			// Import Excel sheet to List
			printf("\nReading file: C:\\Samples\\Tutorial09.xlsx\n");
			EasyXLS::IListPtr  rows = workbook->easy_ReadXLSXSheet_AsList_3("C:\\Samples\\Tutorial09.xlsx", "First tab");
		
			// Confirm import of Excel file
			_bstr_t sError = workbook->easy_getError();
			if (strcmp(sError, "") == 0){
			
				// Display imported List values
				for ( int rowIndex=0; rowIndex<rows->size(); rowIndex++)
				{
					EasyXLS::IListPtr 	row = (EasyXLS::IListPtr) rows->elementAt(rowIndex);
					for (int cellIndex=0; cellIndex<row->size(); cellIndex++)
					{
						printf("At row %d, column %d the value is '%s'\n", (rowIndex+ 1), (cellIndex+ 1), (LPCSTR)((_bstr_t)row->elementAt(cellIndex)));
					}
				}
				printf("\nPress Enter to exit ...");
			}
			else
			{
				printf("\nError reading file C:\\Samples\\Tutorial09.xlsx %s\n", (LPCSTR)sError); 
			}
			
			// Dispose memory
			workbook->Dispose();
		}
		else{
			printf("Object is not available!");
		}
	}
	else{
		printf("COM can't be initialized!");
	}

	// Uninitialize COM
	CoUninitialize();

	_getch();
	return 0;
}

